editor/src/components/editor/editor-component.tsx

```diff
@@ -517,20 +442,74 @@ const OpenFileEditor = betterReactMemo('OpenFileEditor', (props: OpenFileEditorP
     }
   })
 
+  const [runtimeErrors, setRuntimeErrors] = React.useState<Array<RuntimeErrorInfo>>(EmptyArray)
```

Comment:
could you take this code big chunk of code and turn it into a hook? I feel bad for increasing this function's size by 50 lines here